### PR TITLE
Vode magnesium pile fix

### PR DESCRIPTION
**Pull request by vode-code**
## About the PR 
This PR makes magnesium pooling no longer break magnesium's functionality.

## Why's this needed? 
Magnesium being broken is bad.

### CODE REVIEW

**Comment by ZeWaka** [comment]
this has code from your fire breath PR it looks like, please fix.

**Comment by vode-code** [comment]
cannot, this is a github bug and I don't know how to un-commit.

**Comment by ZeWaka** [comment]
> cannot, this is a github bug and I don't know how to un-commit.

it is not a github bug, it's a mistake on your end with git.

File: code/modules/status_system/statusFoodBuffs.dm [hunk 1]
```diff
@@ -302,7 +302,10 @@
 	proc/cast()
 
 		var/turf/T = get_step(owner,owner.dir)
-		T = get_step(T,owner.dir)
+		var/range_breath = 1
+		while((get_dist(owner,T) < range) && (range_breath < 20))// range is used for the range the fireburp can reach from the caster.
+			T = get_step(T,owner.dir)
+			range_breath ++ //range_breath is used to make sure the loop doesn't stay active too long and lag the game if something messes up range.
```
Comment (ZeWaka) [inline code/modules/status_system/statusFoodBuffs.dm:308]:
```suggestion
		T = get_step(T,owner.dir)
```

File: code/modules/status_system/statusFoodBuffs.dm [hunk 2]
```diff
@@ -411,4 +414,3 @@
 					var/turf/T = get_turf(owner)
 					T.fluid_react_single("water",5)
 		return
```
Comment (ZeWaka) [inline code/modules/status_system/statusFoodBuffs.dm:416]:
```suggestion
		return

```

File: code/obj/decal/cleanable.dm [hunk 3]
```diff
@@ -1664,6 +1664,10 @@ var/list/blood_decal_violent_icon_states = list("floor1", "floor2", "floor3", "f
 				else
 					return
 				burn_time--
+			if (on_fire)
+				overlays -= on_fire
+				on_fire = null
+				burn_time = initial(burn_time)
```
Comment (ZeWaka) [inline code/obj/decal/cleanable.dm:1670]:
```suggestion

			if (on_fire)
				overlays -= on_fire
				on_fire = null
				burn_time = initial(burn_time)
```

**Pull request merged**